solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/inbound/acknowledge/JCSMPAcknowledgementCallback.java

```diff
@@ -80,24 +75,12 @@ boolean republishToErrorQueue() {
       return false;
     }
 
-    if (logger.isDebugEnabled()) {
-      logger.debug(String.format("%s %s: Will be republished onto error queue %s",
-          XMLMessage.class.getSimpleName(), messageContainer.getMessage().getMessageId(),
-          errorQueueInfrastructure.getErrorQueueName()));
-    }
+    LOGGER.debug("{} {}: Will be republished onto error queue {}",
+            XMLMessage.class.getSimpleName(), messageContainer.getMessage().getMessageId(),
+            errorQueueInfrastructure.getErrorQueueName());
 
      try {
-       //Check to prevent message from publishing to errorQueue and also redelivered by broker
-       if (messageContainer.isStale()) {
-         throw new IllegalStateException(
-             String.format("Cannot republish failed message container %s " +
-                     "(XMLMessage %s) to error queue %s. Message is stale and will be redelivered.",
-                 messageContainer.getId(), messageContainer.getMessage().getMessageId(),
-                 errorQueueInfrastructure.getErrorQueueName()), null);
-       }
-
-      errorQueueInfrastructure.createCorrelationKey(messageContainer, flowReceiverContainer)
-          .handleError();
+      errorQueueInfrastructure.createCorrelationKey(messageContainer, flowReceiverContainer).handleError();
```

Comment:
ErrorQueueRepublishCorrelationKey might not be a good name, most of the functionality belongs into  ErrorQueueInfrastructure

Reply:
It's called a "correlation key", because that class is eventually used as a correlation key for the republished message: https://github.com/SolaceProducts/solace-spring-cloud/blob/master/solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/util/ErrorQueueInfrastructure.java#L32

Reply:
in this case it's functionality like handling an error may belong into the infrastructure

Reply:
Yeah, I was thinking of maybe combining the two classes. I'm starting to find it a bit weird that they're separate objects.

Will consider doing that in my next PR which will be adding support for error queues in transacted consumers.